### PR TITLE
feat(html): support custom meta element creation

### PR DESCRIPTION
**Pull request by mseeley**
This allows for convenient creation of `meta` elements without reimplementing all template function responsibilities.

## Rollup Plugin Name: `html`

This PR contains:

- [ ] bugfix
- [x] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [x] yes (_bugfixes and features will not be merged without tests_)
- [ ] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers: n/a

### Description

The `html` plugin doesn't allow for injection of `meta` elements without re-implementing all of the template function responsibilities. This is bunch of relative complexity to take on in order to, in my case, inject our old friendly viewport meta below:

```
<meta
  name="viewport"
  content="minimum-scale=1, initial-scale=1, width=device-width"
/>
```

Instead I added support for a backwards compatible `meta` option.
- when omitted the current `utf-8` charset meta will be inserted.
- when provided the charset meta must be supplied manually. To me this was the simplest implementation and an easy to doc/grok all or nothing approach.

~~Running tests locally some snapshots are failing with `null` vs `undefined` values for `map`. I'm running 12.15.0 using nvm. Perhaps there's a behavior changes between `12` and `12.xx`. Tests passed for changed code, so, here we go. :tada:~~

### CODE REVIEW

File: packages/html/README.md [hunk 1]
```diff
@@ -62,6 +62,13 @@ _Note: If using the `es` / `esm` output format, `{ type: 'module'}` is automatic
 Type: `String`<br>
 Default: `'index.html'`
 
+### `meta`
+
+Type: `Array`<br>
+Default: `[{ charset: 'utf-8' }]`
+
+Specifies attributes used to create `meta` elements. For each array member, provide an object with key-value pairs that represent meta element attribute names and values.
+
```
Comment (mseeley) [inline packages/html/README.md:71]: I'm happy to copy/paste any documentation suggestions.

**Comment by mseeley** [comment]
Ah, I'm seeing CI fail with similar problems I had locally. I don't believe my changes are causing those failures.

**Comment by shellscape** [comment]
Thanks for opening this PR. We'll get to reviewing this soon!

**Comment by mseeley** [comment]
@shellscape thanks for updating the `html` snapshots in c09509f8. Diff on this PR is now super tight. 😄

**Comment by mseeley** [comment]
👋 hi maintainers. Please let me know if there's anything I can do to raise visibility of this PR. I'm hoping you all find it to be a reasonable feature addition. Thanks!

File: packages/html/README.md [hunk 2]
```diff
@@ -62,6 +62,13 @@ _Note: If using the `es` / `esm` output format, `{ type: 'module'}` is automatic
 Type: `String`<br>
 Default: `'index.html'`
 
+### `meta`
+
+Type: `Array`<br>
```
Comment (shellscape) [inline packages/html/README.md:67]:
```suggestion
Type: `Array[...object]`<br>
```

File: packages/html/README.md [hunk 3]
```diff
@@ -62,6 +62,13 @@ _Note: If using the `es` / `esm` output format, `{ type: 'module'}` is automatic
...
+Default: `[{ charset: 'utf-8' }]`
+
+Specifies attributes used to create `meta` elements. For each array member, provide an object with key-value pairs that represent meta element attribute names and values.
```
Comment (shellscape) [inline packages/html/README.md:70]:
```suggestion
Specifies attributes used to create `<meta>` elements. For each array item, provide an object with key-value pairs that represent `<meta>` element attribute names and values.
```

**Comment by shellscape** [comment]
thanks!

**Comment by shellscape** [comment]
@mseeley If you're interested in helping to maintain the repo in the future, please let me know. We're always looking for help.

**Pull request merged**